src/components/balances/token-asset-list-item.tsx

```diff
@@ -1,26 +1,44 @@
-import * as React from 'react';
-import { AddressBalanceResponse } from '@stacks/stacks-blockchain-api-types';
+import React, { useEffect, useState } from 'react';
+import BigNumber from 'bignumber.js';
+import { useAtomValue } from 'jotai/utils';
+import { FungibleTokenMetadata } from '@stacks/blockchain-api-client';
 import { Box, color, DynamicColorCircle, Flex, FlexProps, Stack } from '@stacks/ui';
 import { Caption, Text } from '@components/typography';
 import { TxLink } from '@components/links';
-import { getAssetNameParts } from '@common/utils';
+import { ftDecimals, getAssetNameParts } from '@common/utils';
 import { getTicker } from '@components/tx-events';
+import { apiClientsState } from '@store/api-clients';
 
 interface TokenAssetListItemProps extends FlexProps {
+  balanceOrCount: string;
   token: string;
-  isLast: boolean;
-  balances: AddressBalanceResponse;
-  type: 'non_fungible_tokens' | 'fungible_tokens';
+  tokenType: 'non_fungible_tokens' | 'fungible_tokens';
 }
 
 export const TokenAssetListItem: React.FC<TokenAssetListItemProps> = ({
+  balanceOrCount,
   token,
-  balances,
-  type,
-  isLast,
+  tokenType,
 }) => {
+  const [ftMetadata, setFtMetadata] = useState<FungibleTokenMetadata | undefined>();
+  const { tokensApi } = useAtomValue(apiClientsState);
   const { address, asset, contract } = getAssetNameParts(token);
-  const key = type === 'non_fungible_tokens' ? 'count' : 'balance';
+
+  useEffect(() => {
+    const getFtMetadata = async () => {
```

Comment:
I think this fetch logic is worth extracting and placing it outside the view.

Reply:
I had thought abt it but have to send in the tokensApi as an arg which seemed a bit strange but can do.